go/vt/vttablet/tabletserver/tabletenv/config.go

```diff
@@ -882,6 +892,51 @@ func (c *TabletConfig) Verify() error {
 	return nil
 }
 
+// verifyUnmanagedTabletConfig checks unmanaged tablet related config for sanity
+func (c *TabletConfig) verifyUnmanagedTabletConfig() error {
+	// Skip checks if tablet is not unmanaged
+	if !c.UnmanagedTablet {
+		return nil
+	}
+
+	// Throw error if both host and socket are null
+	if !c.DB.HasGlobalSettings() {
+		return errors.New("no connection parameters specified but unmanaged mode specified")
+	}
+	if c.DB.App.User == "" {
+		return errors.New("database app user not specified")
+	}
+	if c.DB.App.Password == "" {
+		return errors.New("database app user password not specified")
+	}
+	if !mysqlctl.DisableActiveReparents {
+		return errors.New("fixing replication should be disabled on unmanaged tablets")
+	}
+
+	return c.checkConnectionForExternalMysql()
+}
+
+// Test connectivity of external mysql
+func (c *TabletConfig) checkConnectionForExternalMysql() error {
+	params := mysql.ConnParams{
+		Host:       c.DB.Host,
+		Port:       c.DB.Port,
+		DbName:     c.DB.DBName,
+		Uname:      c.DB.App.User,
+		Pass:       c.DB.App.Password,
+		UnixSocket: c.DB.Socket,
```

Comment:
Why are we specifying host/port and also socket? You are supposed to need only one of the two.